### PR TITLE
Fix: Plugin installation

### PR DESCRIPTION
**Pull request by josemarinas**
## Description

Fix plugin installation

Task: [OS-302](https://aragonassociation.atlassian.net/browse/OS-302)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder of the package after the [UPCOMING] title and before the latest version.
- [ ] I have tested my code on the test network.

[OS-302]: https://aragonassociation.atlassian.net/browse/OS-302?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: modules/client/examples/05-encoders-decoders/12-apply-installation.ts [hunk 1]
```diff
@@ -46,32 +46,44 @@ const applyInstallationParams: ApplyInstallationParams = {
 
 const daoAddressOrEns: string = "0x123123123123123123123123123123123123"; // "my-dao.eth"
 
-const action: DaoAction = client.encoding.applyInstallationAction(
+const actions: DaoAction[] = client.encoding.applyInstallationAction(
   daoAddressOrEns,
   applyInstallationParams
 );
-console.log({ action });
+console.log({ actions });
```
Comment (brickpop) [inline modules/client/examples/05-encoders-decoders/12-apply-installation.ts]:
```suggestion
console.log(actions);
```

File: modules/client/examples/05-encoders-decoders/12-apply-installation.ts [hunk 2]
```diff
@@ -46,32 +46,44 @@ const applyInstallationParams: ApplyInstallationParams = {
...
 
 /* MARKDOWN
 Returns:
```
Comment (brickpop) [inline modules/client/examples/05-encoders-decoders/12-apply-installation.ts]: Returns three actions:
- Grant root permission to the Plugin Setup Processor
- Ask it to apply the installation
- Revoke the root permission to the Plugin Setup Processor

File: modules/client/src/internal/client/encoding.ts [hunk 3]
```diff
@@ -53,22 +57,47 @@ export class ClientEncoding extends ClientCore implements IClientEncoding {
...
       throw new InvalidAddressError();
     }
+    const provider = this.web3.getProvider();
+    if (!provider) {
+      throw new NoProviderError();
+    }
+    const network = provider.network.name as SupportedNetworks;
+    if (!SupportedNetworksArray.includes(network)) {
+      throw new UnsupportedNetworkError(network);
+    }
```
Comment (brickpop) [inline modules/client/src/internal/client/encoding.ts:71]: Didn't we have a way to encapsulate this?

File: modules/client/test/integration/client/decoding.test.ts [hunk 4]
```diff
@@ -522,12 +522,12 @@ describe("Client", () => {
         pluginAddress: "0x1234567890123456789012345678901234567890",
       };
 
-      const action = client.encoding.applyInstallationAction(
+      const actions = client.encoding.applyInstallationAction(
         "0x1234567890123456789012345678901234567890",
         applyInstallationParams,
       );
       const decodedApplyInstallationParams = client.decoding.applyInstallationAction(
```
Comment (brickpop) [inline modules/client/test/integration/client/decoding.test.ts:536]:
```suggestion
      expect(actions.length).toBe(3);

      const decodedApplyInstallationParams = client.decoding.applyInstallationAction(
```

File: modules/client/test/integration/client/encoding.test.ts [hunk 5]
```diff
@@ -390,16 +390,16 @@ describe("Client", () => {
...
       const daoAddress = "0x1234567890123456789012345678901234567890";
-      const action = client.encoding.applyInstallationAction(
+      const actions = client.encoding.applyInstallationAction(
         daoAddress,
         applyInstallationParams,
       );
 
-      expect(typeof action).toBe("object");
-      expect(action.data).toBeInstanceOf(Uint8Array);
+      expect(typeof actions[1]).toBe("object");
```
Comment (brickpop) [inline modules/client/test/integration/client/encoding.test.ts:405]:
```suggestion
      expect(actions.length).toBe(3);
      expect(typeof actions[1]).toBe("object");
```

File: modules/client/test/integration/client/decoding.test.ts [hunk 6]
```diff
@@ -28,8 +29,13 @@ import {
 } from "../../../src/interfaces";
 import { bytesToHex, hexToBytes } from "@aragon/sdk-common";
 import { defaultAbiCoder } from "@ethersproject/abi";
+import { JsonRpcProvider } from "@ethersproject/providers";
+import { AddressZero } from "@ethersproject/constants";
 
 describe("Client", () => {
+  beforeAll(async () => {
```
Comment (brickpop) [inline modules/client/test/integration/client/decoding.test.ts]: Why this line?

File: modules/client/src/internal/client/encoding.ts (hunk 3 above)
Comment (josemarinas) [reply modules/client/src/internal/client/encoding.ts:71]: nope

**Review by brickpop — APPROVED** [review]

**Review by brickpop — APPROVED** [review]

**Pull request merged**